exporter/prometheusexporter/metric_rename.go

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter"
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+var UnitMap = map[string]string{
+	"s":   "seconds",
+	"Cel": "celsius",
+	"m":   "meters",
+	"By":  "bytes",
+	"V":   "volts",
+	"A":   "amperes",
+	"J":   "joules",
+	"W":   "watts",
+	"g":   "grams",
+	"Hz":  "hertz",
+	"$":   "dollars",
+	"1":   "ratio",
+}
+
+// Build a Prometheus-compliant metric name for the specified metric
+func buildPromCompliantName(metric pmetric.Metric) string {
+
+	name := sanitize(strings.TrimSpace(metric.Name()), true)
+	unit := strings.TrimSpace(metric.Unit())
+	isCounter := metric.DataType() == pmetric.MetricDataTypeSum && metric.Sum().IsMonotonic()
+
+	// Remove _total suffix for now
+	// The _total suffix will be appended later for Counters
+	if isCounter {
+		name = strings.TrimSuffix(name, "_total")
+	}
+
+	if promStandardUnit, ok := UnitMap[unit]; ok {
+
+		// Standard Otel unit is converted to standard Prom unit
+		promStandardUnit = "_" + promStandardUnit
+		if !strings.HasSuffix(name, promStandardUnit) {
+			name = name + promStandardUnit
+		}
+
+	} else if !strings.HasPrefix(unit, "{") || !strings.HasSuffix(unit, "}") {
+
+		// Unit is not Prometheus-standard and is not surrounded with {}
+		// We will append it after sanitization
+		promTentativeUnit := strings.ReplaceAll(unit, "/", "_per_")
+		promTentativeUnit = "_" + sanitize(strings.ToLower(promTentativeUnit), true)
+		if strings.HasSuffix(promTentativeUnit, "_per_s") {
+			promTentativeUnit = promTentativeUnit + "econd"
```

Comment:
If we aren't going to do this for all known units, can we omit this for now?

Reply:
Yep, I knew you wouldn't like this 😁